utils/config_file.hh

```diff
@@ -66,9 +66,15 @@ public:
     typedef std::vector<sstring> string_list;
 
     enum class value_status {
-        Used,
-        Unused,
-        Invalid,
+        Used, ///< supported by Scylla. only the "Used" options are added to the
+              ///< command line options, and can be specified with command line.
+        Unused, ///< no-more supported. we print a message at seeing each of
+                ///< these options. typically used for options supported by an
+                ///< older version of Scylla.
+        Invalid, ///< only supported by Cassandra. we just silently ignore
+                 ///< these options at seeing them. these options are kept
+                 ///< around only to ease the migration process so user can
+                 ///< keep using their existing settings.
```

Comment:
Are you sure you didn't reverse the explanation of Unused and Invalid?
As far as I can see in utils/config_file.cc, using an Invalid option in yaml leads to an error "Option is not applicable", while "Unused" is silently accepted as saved (it is saved in memory, but since nobody will use this configuration variable later, it is virtually ignored).

Reply:
@nyh thank you for your review! indeed, i managed to put the transpose these two comments. should be fixed now.